cql3/expr/term_expr.cc

```diff
@@ -0,0 +1,1004 @@
+            }
+            break;
+        case untyped_constant::type_class::floating_point:
+            if (cql3_type::kind_enum_set::frozen<
+                    cql3_type::kind::DECIMAL,
+                    cql3_type::kind::DOUBLE,
+                    cql3_type::kind::FLOAT>::contains(kind)) {
+                return assignment_testable::test_result::WEAKLY_ASSIGNABLE;
+            }
+            break;
+        case untyped_constant::type_class::boolean:
+            if (kind == cql3_type::kind_enum_set::prepare<cql3_type::kind::BOOLEAN>()) {
+                return assignment_testable::test_result::WEAKLY_ASSIGNABLE;
+            }
+            break;
+        case untyped_constant::type_class::hex:
+            if (kind == cql3_type::kind_enum_set::prepare<cql3_type::kind::BLOB>()) {
+                return assignment_testable::test_result::WEAKLY_ASSIGNABLE;
+            }
+            break;
+        case untyped_constant::type_class::duration:
+            if (kind == cql3_type::kind_enum_set::prepare<cql3_type::kind::DURATION>()) {
+                return assignment_testable::test_result::EXACT_MATCH;
+            }
+            break;
+    }
+    return assignment_testable::test_result::NOT_ASSIGNABLE;
+}
+
+static
+::shared_ptr<term>
+untyped_constant_prepare_term(const untyped_constant& uc, database& db, const sstring& keyspace, const column_specification_or_tuple& receiver_)
+{
+    auto& receiver = std::get<lw_shared_ptr<column_specification>>(receiver_);
+    if (!is_assignable(untyped_constant_test_assignment(uc, db, keyspace, *receiver))) {
+        throw exceptions::invalid_request_exception(format("Invalid {} constant ({}) for \"{}\" of type {}",
+            uc.partial_type, uc.raw_text, *receiver->name, receiver->type->as_cql3_type().to_string()));
+    }
+    return ::make_shared<constants::value>(cql3::raw_value::make_value(untyped_constant_parsed_value(uc, receiver->type)));
+}
+
+static
+assignment_testable::test_result
+bind_variable_test_assignment(const bind_variable& bv, database& db, const sstring& keyspace, const column_specification& receiver) {
+    return assignment_testable::test_result::WEAKLY_ASSIGNABLE;
+}
+
+static
+::shared_ptr<term>
+bind_variable_scalar_prepare_term(const bind_variable& bv, database& db, const sstring& keyspace, const column_specification_or_tuple& receiver_)
+{
+    auto& receiver = std::get<lw_shared_ptr<column_specification>>(receiver_);
+    if (receiver->type->is_collection()) {
+        if (receiver->type->without_reversed().is_list()) {
+            return ::make_shared<lists::marker>(bv.bind_index, receiver);
+        } else if (receiver->type->without_reversed().is_set()) {
+            return ::make_shared<sets::marker>(bv.bind_index, receiver);
+        } else if (receiver->type->without_reversed().is_map()) {
+            return ::make_shared<maps::marker>(bv.bind_index, receiver);
+        }
+        assert(0);
+    }
+
+    if (receiver->type->is_user_type()) {
```

Comment:
looks like it shoud be `receiver->type->without_reversed().is_user_type()`

Reply:
Copied verbatim from old code.  @cvybhu can you please check if master has a bug here?

Reply:
The same code is on master. It looks like a bug, but maybe it works somehow, will have to check.